laws/src/main/scala/cats/laws/DecideableLaws.scala

```diff
@@ -0,0 +1,28 @@
+package cats
+package laws
+
+trait DecideableLaws[F[_]] extends ContravariantMonoidalLaws[F] {
+  implicit override def F: Decideable[F]
+
+  def decideableDecideRightAbsorption[A](fa: F[A]): IsEq[F[A]] =
```

Comment:
This one tests right identity, not absorption, right?
Absorption would be `product(fa, zero) == zero`

Reply:
For `ContravariantMonoidal` we already have left and right multiplicative identity as well as associativity.
So for `Decidable` I think we need :

1. `sum` and `zero` Left Identity
2. `sum` and `zero` Right Identity
3. `sum` Associativity
4. `product` and `zero` Right absorption
5. `sum` and `contramap` Right Distributivity
6. `sum` and `product` Right Distributivity

And then we should be good IMO :)

Reply:
Ah I see how you'd write absorption I think now, yeah, sounds good.

Reply:
Interestingly, having both left and right identities seems to rule out `OptionT` because you have to make _a_ choice for `None`, and similarly the sum and product distribution rules out `Order` and `Ordering` instances as you end up reversing orderings in the course of that, though I'm working to see if I can fix those instances to respect it.